test/Microsoft.AspNetCore.Mvc.Core.Test/ContentResultTest.cs

```diff
@@ -167,6 +138,66 @@ public class ContentResultTest
             var finalResponseContentType = httpContext.Response.ContentType;
             Assert.Equal(expectedContentType, finalResponseContentType);
             Assert.Equal(expectedContentData, memoryStream.ToArray());
+            Assert.Equal(expectedContentData.Length, httpContext.Response.ContentLength);
+        }
+
+        public static TheoryData<string, string> ContentResult_WritesDataCorrectly_ForDifferentContentSizesData
+        {
+            get
+            {
+                var maxCharacterChunkSize = ContentResultExecutor.MaxCharacterChunkSize;
+
+                // content, contentType
+                return new TheoryData<string, string>
+                {
+                    //色
+                    {  string.Empty, "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize - 1), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize + 1), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize - 2), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize + 2), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize - 3), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize + 3), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize * 2), "text/plain; charset=utf-8" },
+                    {  new string('a', maxCharacterChunkSize * 3), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize - 1), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize + 1), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize - 2), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize + 2), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize - 3), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize + 3), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize * 2), "text/plain; charset=utf-8" },
+                    {  new string('色', maxCharacterChunkSize * 3), "text/plain; charset=utf-8" }
```

Comment:
I'd also like to see a few more tests with multiples of the chunk size **and** a few characters +/-